src/SettingsMigrator/Migrator.cs

```diff
@@ -0,0 +1,276 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Xml;
+
+namespace Microsoft.VisualStudio.TestPlatform.CommandLine
+{
+    public class Migrator
+    {
+        /// <summary>
+        /// Given a runsettings with an embedded testsettings, converts it to runsettings.
+        /// </summary>
+        /// <param name="oldRunsettingsPath"></param>
+        /// <param name="newRunsettingsPath"></param>
+        public void MigrateRunsettings(string oldRunsettingsPath, string newRunsettingsPath)
+        {
+            string testsettingsPath = null;
+            using (XmlTextReader reader = new XmlTextReader(oldRunsettingsPath))
+            {
+                reader.Namespaces = false;
+
+                var document = new XmlDocument();
+                document.Load(reader);
+                var root = document.DocumentElement;
+
+                var testsettingsNode = root.SelectSingleNode(@"/RunSettings/MSTest/SettingsFile");
+
+                if (testsettingsNode != null)
+                {
+                    testsettingsPath = testsettingsNode.InnerText;
+                }
+                if (!string.IsNullOrWhiteSpace(testsettingsPath))
+                {
+                    // Expand path relative to runsettings location.
+                    if (!Path.IsPathRooted(testsettingsPath))
+                    {
+                        testsettingsPath = Path.Combine(oldRunsettingsPath, testsettingsPath);
+                    }
+
+                    MigrateTestsettings(testsettingsPath, newRunsettingsPath, File.ReadAllText(oldRunsettingsPath));
+                }
+                else
+                {
+                    Console.WriteLine("Runsettings does not contain an embedded testsettings, not migrating.");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Given a testsettings, converts it to runsettings
+        /// </summary>
+        /// <param name="testsettingsPath"></param>
+        /// <param name="newRunsettingsPath"></param>
+        /// <param name="oldRunSettingsContent"></param>
+        public void MigrateTestsettings(string testsettingsPath, string newRunsettingsPath, string oldRunSettingsContent = null)
```

Comment:
Let's refactor this to create file separately, and have a common api for translating testsettings xml to required runsettings xml
MigrateLegacyNodesToRunSettings

Reply:
Done